services/provider/server/server.go

```diff
@@ -1166,13 +1167,39 @@ func (s *OCSProviderServer) getDesiredCephConnection(
 		return nil, fmt.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
 	}
 
+	readAffinityOptions := util.GetReadAffinityOptions(storageCluster)
+	readAffinityDisabledAnnotation, _ := strconv.ParseBool(consumer.Annotations["ocs.openshift.io/disable-read-affinity"])
+
+	var readAffinity *csiopv1a1.ReadAffinitySpec = nil
+	if readAffinityOptions.Enabled && !readAffinityDisabledAnnotation {
+		labels := readAffinityOptions.CrushLocationLabels
+		if len(labels) == 0 {
+			labels = []string{
+				"topology.kubernetes.io/region",
+				"topology.kubernetes.io/zone",
+				"topology.rook.io/datacenter",
+				"topology.rook.io/room",
+				"topology.rook.io/pod",
+				"topology.rook.io/pdu",
+				"topology.rook.io/row",
+				"topology.rook.io/rack",
+				"topology.rook.io/chassis",
+				"kubernetes.io/hostname",
```

Comment:
Hostname 
Zone
Region
Chassis 
Rack
....


This is the correct order try to Match with Rook code
/hold

Reply:
@OdedViner The order in the code is still not aligned with the order @parth-gr provided above. Can you please fix it?

Reply:
Cc @Rakshith-R ^^

Reply:
@parth-gr  can you check now? https://github.com/rook/rook/blob/cd2d887c928821e95b6d008981bc2826a6a32083/pkg/operator/ceph/csi/cluster_config_test.go#L434